internal/pluginframework/tfschema/struct_to_schema_test.go

```diff
@@ -0,0 +1,152 @@
+package tfschema
+
+import (
+	"context"
+	"fmt"
+	"testing"
+
+	"github.com/hashicorp/terraform-plugin-framework/tfsdk"
+	"github.com/hashicorp/terraform-plugin-framework/types"
+	"github.com/stretchr/testify/assert"
+)
+
+type TestStringTfSdk struct {
+	Description types.String `tfsdk:"description" tf:"optional"`
+}
+
+type TestBoolTfSdk struct {
+	Enabled types.Bool `tfsdk:"enabled" tf:""`
+}
+
+type TestIntTfSdk struct {
+	Workers types.Int64 `tfsdk:"workers" tf:"optional"`
+}
+
+type TestFloatTfSdk struct {
+	Float types.Float64 `tfsdk:"float" tf:"optional"`
+}
+
+type TestListTfSdk struct {
+	Repeated []types.Int64 `tfsdk:"repeated" tf:"optional"`
+}
+
+type TestMapTfSdk struct {
+	Attributes map[string]types.String `tfsdk:"attributes" tf:"optional"`
+}
+
+type TestNestedListTfSdk struct {
+	NestedList []DummyNested `tfsdk:"nested_list" tf:"optional"`
+}
+
+type DummyNested struct {
+	Name    types.String `tfsdk:"name" tf:"optional"`
+	Enabled types.Bool   `tfsdk:"enabled" tf:"optional"`
+}
+
+type TestNestedMapTfSdk struct {
+	NestedMap map[string]DummyNested `tfsdk:"nested_map" tf:"optional"`
+}
+
+type TestPointerTfSdk struct {
+	Nested *DummyNested `tfsdk:"nested" tf:"optional"`
+}
+
+var tests = []struct {
```

Comment:
Can we add a couple more test cases:
* Input is a pointer to a struct
* Input has a field that is a pointer to a struct

Reply:
Sounds good, I think the case for "input is a pointer to a struct" already exists